File: festival-kotlin-web/src/main/java/cn/bdqfork/web/annotation/RolesAllowed.java
package cn.bdqfork.web.annotation;

import cn.bdqfork.web.constant.LogicType;

import java.lang.annotation.*;

/**
 * 表示哪些角色可以访问api
 *
 * @author bdq
 * @since 2020/1/27
 */
@Auth
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface RolesAllowed {
    String[] value();

    LogicType logic() default LogicType.AND;
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/annotation/RolesAllowed.kt
package cn.bdqfork.web.annotation

import cn.bdqfork.web.constant.LogicType

/**
 * 表示哪些角色可以访问api
 *
 * @author bdq
 * @since 2020/1/27
 */
@Auth
@MustBeDocumented
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.FUNCTION)
annotation class RolesAllowed(vararg val value: String, val logic: LogicType = LogicType.AND)